java/sidecar/src/main/java/com/instaclustr/cassandra/sidecar/operation/OperationTask.java

```diff
@@ -0,0 +1,111 @@
+package com.instaclustr.cassandra.sidecar.operation;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+
+import com.instaclustr.cassandra.sidecar.model.operation.OperationState;
+import com.instaclustr.cassandra.sidecar.model.operation.Operation;
+import com.instaclustr.cassandra.sidecar.model.result.OperationResult;
+import jmx.org.apache.cassandra.service.StorageServiceMBean;
+
+public abstract class OperationTask<O extends Operation, R extends OperationResult> implements Callable<R> {
+
+    private final O operation;
+
+    private final R result;
+
+    private StopWatch stopWatch = new StopWatch();
+
+    protected StorageServiceMBean storageServiceMBean;
```

Comment:
A reference to the StorageService in the base class shouldn't be necessary. Look into Guice's AssistedInject.

https://github.com/google/guice/wiki/AssistedInject